src/main/java/dog/pawbook/model/dog/Dog.java

```diff
@@ -0,0 +1,134 @@
+package dog.pawbook.model.dog;
+
+import static dog.pawbook.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.text.ParseException;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import dog.pawbook.model.owner.Name;
+import dog.pawbook.model.owner.Owner;
+import dog.pawbook.model.tag.Tag;
+
+
+/**
+ * Represents a dog in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Dog {
+
+    enum Sex {
+        MALE,
+        FEMALE
+    }
+
+    //Identity Fields
+    private final Name name;
+    private final Breed breed;
+    private final DateOfBirth dob;
+    private final Sex sex;
+
+    // Data fields
+    private final Owner owner;
+    private final Set<Tag> tags = new HashSet<>();
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Dog(Name name, Breed breed, DateOfBirth dob, Sex sex, Owner owner, Set<Tag> tags) {
+        requireAllNonNull(name, breed, dob, sex, owner, tags);
+        this.name = name;
+        this.breed = breed;
+        this.dob = dob;
+        this.sex = sex;
+        this.owner = owner;
+        this.tags.addAll(tags);
+    }
+
+    public Name getName() {
+        return name;
+    }
+
+    public Breed getBreed() {
+        return breed;
+    }
+
+    public DateOfBirth getDob() {
+        return dob;
+    }
+
+    public int getAge() throws ParseException {
+        return dob.getAge(dob.value);
+    }
+
+
+
+    public Sex getSex() {
+        return sex;
+    }
+
+    public Owner getOwner() {
+        return owner;
+    }
+
+    /**
+     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    public Set<Tag> getTags() {
+        return Collections.unmodifiableSet(tags);
+    }
+
+
+    /**
+     * Returns true if both owners have the same identity and data fields.
+     * This defines a stronger notion of equality between two owners.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Dog)) {
+            return false;
+        }
+
+        Dog otherDog = (Dog) other;
+        return otherDog.getName().equals(getName())
+            && otherDog.getBreed().equals(getBreed())
+            && otherDog.getSex().equals(getSex())
+            && otherDog.getDob().equals(getDob());
```

Comment:
Is the indentation here supposed to be 8 spaces? Can consider updating your IDE settings to automatically indent by 8 spaces.